Dockerfile

```diff
@@ -1,42 +1,45 @@
 # Stage 1: Install Python dependencies
 FROM python:3.13-slim AS python-builder
+
 WORKDIR /build
 
-# System build tools
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends gcc python3-dev \
-    && rm -rf /var/lib/apt/lists/*
+RUN set -eux; \
+    apt-get update; \
+    apt-get install -y --no-install-recommends gcc python3-dev; \
+    rm -rf /var/lib/apt/lists/*
 
-# Metadata and code that setuptools needs
 COPY pyproject.toml .
 COPY src/ ./src/
```

Comment:
I don't like this but we can't leverage docker layer caching and install dependencies first
See #398